dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -37,148 +36,128 @@
-
-					return Response.status(HttpStatus.SC_OK).build();
-				} catch (NumberFormatException e) {
-				    try {
-		                HibernateUtil.rollbackTransaction();
-		            } catch (DotHibernateException e1) {
-		                Logger.error(this, "error rollback",e1);
-		            }
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				} catch (Exception e) {
-				    try {
-		                HibernateUtil.rollbackTransaction();
-		            } catch (DotHibernateException e1) {
-		                Logger.error(this, "error rollback",e1);
-		            }
-					Logger.error(PublisherQueueJob.class, "Error caused by remote call of: "+remoteIP);
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				}
-				finally {
-				    try {
-		                HibernateUtil.closeSession();
-		            } catch (DotHibernateException e) {
-		                Logger.error(this, "error close session",e);
-		            }
-				}
-    		}
-    	} catch (IOException e) {
-    		Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-    	}
-
-		return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).build();
+			@QueryParam("type")        final String type,
+			@QueryParam("callback")    final String callback,
+			@QueryParam("FORCE_PUSH")  final boolean forcePush,
+			@Context final HttpServletRequest  request,
+			@Context final HttpServletResponse response
+	) throws Exception {
+		final ResourceResponse responseResource = new ResourceResponse(
+				CollectionsUtils.map("type", type, "callback", callback));
+		final String remoteIP = UtilMethods.isSet(request.getRemoteHost())?
+				request.getRemoteHost() : request.getRemoteAddr();
+
+		if (request.getInputStream().isFinished()) {
+			Logger.error(this.getClass(), "Push Publishing failed from " + remoteIP + " bundle expected");
+			return responseResource.responseError(HttpStatus.SC_BAD_REQUEST);
+		}
+
+		final AuthCredentialPushPublishUtil.PushPublishAuthenticationToken pushPublishAuthenticationToken
+				= AuthCredentialPushPublishUtil.INSTANCE.processAuthHeader(request);
+
+		final Optional<Response> failResponse = PushPublishResourceUtil.getFailResponse(remoteIP, pushPublishAuthenticationToken);
+
+		if (failResponse.isPresent()) {
+			return failResponse.get();
+		}
+
+		final Bundle bundle = this.publishBundle(forcePush, request, remoteIP);
+
+		return Response.ok(bundle).build();
 	}
 
-    /**
-     * Validates a received token
-     *
-     * @param token    Token to validate
-     * @param remoteIP Sender IP
-     * @param mySelf   Current end point
-     * @return True if valid
-     * @throws IOException If fails reading the security token
-     */
-    public static boolean isValidToken ( String token, String remoteIP, PublishingEndPoint mySelf ) throws IOException {
+	@WrapInTransaction
+	private Bundle publishBundle(final boolean forcePush,
+								 final HttpServletRequest request,
+								 final String remoteIP) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)